packages/session-replay-browser/src/session-replay.ts

```diff
@@ -37,13 +41,22 @@ export class SessionReplay implements AmplitudeSessionReplay {
   }
 
   protected async _init(apiKey: string, options: SessionReplayOptions) {
-    this.config = new SessionReplayConfig(apiKey, options);
-    this.loggerProvider = this.config.loggerProvider;
-    this.identifiers = new SessionIdentifiers(options, this.loggerProvider);
-
-    this.eventsManager = new SessionReplayEventsManager({
-      config: this.config,
+    this.loggerProvider = options.loggerProvider || new Logger();
+    this.sessionIDBStore = new SessionReplaySessionIDBStore({
+      loggerProvider: this.loggerProvider,
+      apiKey: apiKey,
     });
+    this.identifiers = new SessionIdentifiers(options, this.loggerProvider);
+    this.joinedConfigGenerator = new SessionReplayJoinedConfigGenerator(apiKey, options, this.sessionIDBStore);
+    if (this.identifiers.sessionId) {
+      this.config = await this.joinedConfigGenerator.generateJoinedConfig(this.identifiers.sessionId);
+    }
+    if (this.config) {
+      this.eventsManager = new SessionReplayEventsManager({
```

Comment:
thought/not blocking. 

Longer term, this makes me think we should decouple the config & events storage in the key. I.e. `<api-key>/sr/<session-id>/config` for the configs, and `<api-key>/sr/<session-id>/events` for sr events.

Reply:
Definitely - will be doing this in my IDB refactor